apps/sms/views/shared/js/activity_handler.js

```diff
@@ -438,13 +441,19 @@ var ActivityHandler = {
                 text = slideArray[i].text;
                 break;
               }
-              text = text ? text : _('mms-message');
-              callback(text);
+
+              if (text) {
+                defer.resolve(text);
+              } else {
+                formatValue('mms-message').then(defer.resolve);
+              }
             });
           }
+
+          return defer.promise;
         }
 
-        Contacts.findByAddress(message.sender).then(function(contacts) {
+        return Contacts.findByAddress(message.sender).then(function(contacts) {
```

Comment:
Nope, I was asking if we should use this method in real code, especially after reading comment from [1]:

```
// This method can be called only from the systeMessageCallback function and
// it allows the page to set a promise to keep alive the app until the
// current operation is not fully completed.
```

So it's just out of curiosity, maybe we can ask KanRu?